base/strings/basic.jl

```diff
@@ -140,7 +140,7 @@ See also: [`getindex`](@ref), [`start`](@ref), [`done`](@ref),
 
 start(s::AbstractString) = 1
 done(s::AbstractString, i::Integer) = i > ncodeunits(s)
-eltype(::Type{<:AbstractString}) = Char
+eltype(::Type{<:AbstractString}) = Char # some string types may use another AbstractChar
```

Comment:
Is there some way we can put an `::eltype(s)` type assert somewhere to catch this if it's wrong?

Reply:
Not sure…where would it go?